src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunHelperNode.cs

```diff
@@ -133,5 +156,75 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                 return null;
             }
         }
+
+        protected override void OnMarked(NodeFactory factory)
+        {
+            switch (_id)
+            {
+                case ReadyToRunHelperId.GenericLookupFromThis:
+                case ReadyToRunHelperId.GenericLookupFromDictionary:
+                    {
+                        // When the helper call gets marked, ensure the generic layout for the associated dictionaries
+                        // includes the signature.
+                        var lookupInfo = (GenericLookupDescriptor)_target;
+                        factory.GenericDictionaryLayout(lookupInfo.CanonicalOwner).EnsureEntry(lookupInfo.Signature);
```

Comment:
I agree with Jan, I would avoid doing this for now. The sparse handling of virtuals in the compiler is already very expensive, and I'd like not to see that expand to many other places.

---
In reply to: [80807977](https://github.com/dotnet/corert/pull/1933#discussion_r80807977) [](ancestors = 80807977)